forge-game/src/main/java/forge/game/ability/effects/PlaneswalkEffect.java

```diff
@@ -22,11 +23,14 @@ public void resolve(SpellAbility sa) {
                 p.leaveCurrentPlane();
             }
         }
+        if (sa.hasParam("Optional") && sa.getActivatingPlayer().getController().confirmAction(sa, null,
+                Localizer.getInstance().getMessage("lblWouldYouLikeToPlaneswalk"), null)) {
+                    return;
+        }
```

Comment:
We've already left because of the loop starting in line 22

I think you could just put this if statement right after the getActivePlanes if statement